core/Request.php

```diff
@@ -204,17 +203,18 @@ public function setFlashdata($name, $value)
      * @params string $name
      */
 
-    public function getFlashdata($name = false)
+    public function getFlashdata($name = false, $default = false)
     {
-
-        if ($name) {
-            return (!empty($this->flashdata[$name]) ? $this->flashdata[$name]['value'] : false);
-
-        } else {
+        if ($name === false && !empty($this->flashdata)) {
             return $this->flashdata;
-
+        }
+        if($name !== false) {
+            if(!empty($this->flashdata[$name])) {
```

Comment:
You can reduce the cyclomatic complexity of this method by merging this if sttement with the one above it as you did on line 208